lib/routes/email.js

```diff
@@ -33,7 +33,15 @@ module.exports = {
         return reply(new AppError.authError('network error'));
       }
       if (res.statusCode >= 400) {
-        logger.error('request.auth_server.fail', { code: res.statusCode });
+        body = body && body.code ? body : { code: res.statusCode };
+        // It's possible for the account to be deleted on the auth-server
+        // but its oauth tokens are still valid, pending cleanup.
+        // We just treat it as if auth failed.
+        if (body.errno === 102) {
+          logger.info('request.auth_server.fail', body);
+          return reply(new AppError.unauthorized(body.message));
+        }
+        logger.error('request.auth_server.fail', body);
         return reply(new AppError.authError('auth server error'));
```

Comment:
So, this seems wrong even with the change. Any 400 response, except one with subcode 102, is being logged and reported back to the client as a '503 Service Unavailable'. There's no service that is unavailable if a 4xx came back from the auth service.